test-resources/headless-core-stub/lambdas/mock-jwks/src/services/cache-jwk.ts

```diff
@@ -0,0 +1,33 @@
+import { randomUUID } from "crypto";
+import { JSONWebKeySet, JWK } from "jose";
+import { getJwkKeyPair, GetJwkKeyPairOptions } from "../../../../utils/src/keypair";
+import { getHashedKid } from "../../../../utils/src/hashing";
+import { ClientConfiguration } from "../../../../utils/src/services/client-configuration";
+import { CORE_STUB_SIGNING_PUBLIC_JWK } from "../constants";
+
+let cachedKeys: { jwks: JSONWebKeySet } | null = null;
+
+export const generateJWKS = async (clientId: string): Promise<{ jwks: JSONWebKeySet; privateKeys?: JWK[] }> => {
+    if (cachedKeys) {
+        return cachedKeys;
+    }
+
+    const ssmParameters = await ClientConfiguration.getConfig(clientId);
+    const currentPublicKey = CORE_STUB_SIGNING_PUBLIC_JWK as JWK;
+    const currentPrivateKey = JSON.parse(ssmParameters.privateSigningKey) as JWK;
```

Comment:
Sorry I'm really not following what we are doing here with private keys. We seem to add them to an object and then not use them? What am I missing?

Reply:
I use them in the test, as it is a `keypair`, to show `publickey.kid === privatekey.kid`

Reply:
Which test? I wonder if there's a way around it all all? Doesn't feel quite right to give a lambda knowledge of a private key just for testing